code/modules/clothing/head/perceptomatrix.dm

```diff
@@ -0,0 +1,265 @@
+		to_chat(caster, "You can't zap minds without a core installd!")
+		return SPELL_CANCEL_CAST
+
+/obj/item/clothing/head/helmet/perceptomatrix/proc/update_anomaly_state()
+
+	// If the core isn't installed, or it's temporarily deactivated, disable special functions.
+	if(!core_installed)
+		clothing_flags = PERCEPTOMATRIX_INACTIVE_FLAGS
+		detach_clothing_traits(clothing_traits)
+		QDEL_LIST(active_components)
+		return
+
+	clothing_flags = PERCEPTOMATRIX_ACTIVE_FLAGS
+	attach_clothing_traits(initial(clothing_traits))
+
+	LAZYADD(active_components, AddComponent(/datum/component/wearertargeting/earprotection, list(ITEM_SLOT_HEAD)))
+	LAZYADD(active_components, AddComponent(
+		/datum/component/anti_magic, \
+		antimagic_flags = MAGIC_RESISTANCE_MIND, \
+		inventory_flags = ITEM_SLOT_HEAD, \
+	))
+
+	update_icon_state()
+
+/obj/item/clothing/head/helmet/perceptomatrix/Destroy(force)
+	QDEL_NULL(active_components)
+	return ..()
+
+/obj/item/clothing/head/helmet/perceptomatrix/examine(mob/user)
+	. = ..()
+	if (!core_installed)
+		. += span_warning("It requires a hallucination anomaly core in order to function.")
+
+/obj/item/clothing/head/helmet/perceptomatrix/item_action_slot_check(slot, mob/user, datum/action/action)
+	return slot & ITEM_SLOT_HEAD
+
+/obj/item/clothing/head/helmet/perceptomatrix/update_icon_state()
+	icon_state = base_icon_state + (core_installed ? "" : "_inactive")
+	worn_icon_state = base_icon_state + (core_installed ? "" : "_inactive")
+	return ..()
+
+/obj/item/clothing/head/helmet/perceptomatrix/item_interaction(obj/item/weapon, mob/user, params)
+	if (!istype(weapon, /obj/item/assembly/signaler/anomaly/hallucination))
+		return NONE
+	balloon_alert(user, "inserting...")
+	if (!do_after(user, delay = 3 SECONDS, target = src))
+		return ITEM_INTERACT_BLOCKING
+	qdel(weapon)
+	core_installed = TRUE
+	update_anomaly_state()
+	update_appearance(UPDATE_ICON_STATE)
+	playsound(src, 'sound/machines/crate/crate_open.ogg', 50, FALSE)
+	return ITEM_INTERACT_SUCCESS
+
+/obj/item/clothing/head/helmet/perceptomatrix/functioning
+	core_installed = TRUE
+
+/datum/action/cooldown/spell/pointed/percept_hallucination
+	name = "Hallucinate"
+	desc = "Redirect perceptual energies towards a target, staggering them."
+	button_icon_state = "blind"
+	ranged_mousepointer = 'icons/effects/mouse_pointers/blind_target.dmi'
+
+	sound = 'sound/items/weapons/emitter2.ogg'
+	school = SCHOOL_PSYCHIC
+	cooldown_time = 15 SECONDS
+
+	invocation_type = INVOCATION_NONE
+	spell_requirements = NONE
+	antimagic_flags = MAGIC_RESISTANCE_MIND
+
+	active_msg = "You prepare to zap a target with hallucinations..."
+
+	/// The amount of blurriness to apply
+	var/eye_blur_duration = 7 SECONDS
+	/// The amount of stagger to apply
+	var/stagger_duration = 3 SECONDS
+	/// The amount of hallucination to apply
+	var/hallucination_duration = 25 SECONDS
+	/// Helmet the spell's bound to.
+	var/obj/item/clothing/head/helmet/perceptomatrix/linked_helmet
```

Comment:
Just dump this var now you shouldn't need it anymore (and it seems unused anyways)